core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -2205,18 +2207,15 @@ private void mountInternal(LockedInodePath inodePath, AlluxioURI ufsPath, boolea
       // Check that the ufsPath exists and is a directory
       UnderFileSystem ufs = UnderFileSystem.get(ufsPath.toString());
       ufs.setProperties(options.getProperties());
-      if (!ufs.exists(ufsPath.toString())) {
+      if (!ufs.isDirectory(ufsPath.toString())) {
```

Comment:
Is there any way to recover the lost bit of information (failure due to not exist or is file)? Alternatively we can just combine the two errors into one.